grails-app/services/com/unifina/service/CanvasService.groovy

```diff
@@ -201,8 +202,50 @@ class CanvasService {
 		return grailsLinkGenerator.link(controller: 'canvas', action: 'editor', id: canvas.id, absolute: true)
 	}
 
+	@CompileStatic
+	def addExampleCanvases(SecUser user, List<Canvas> examples) {
+		for (Canvas c : examples) {
+			switch (c.exampleType) {
+				// Create a copy of the example canvas for the user and grant all permissions.
+				case ExampleType.COPY:
+					Canvas canvas = new Canvas()
+					canvas.dateCreated = c.dateCreated
```

Comment:
This code will be difficult to maintain if we enumerate all the fields explicitly. In groovy you can do something like `canvas.properties = c.properties` to do it via reflection. Would that work for copying most of the fields? Then you can overwrite the necessary fields explicitly (like `canvas.exampleType = ExampleType.NOT_SET`)